mithril/mithril.d.ts

```diff
@@ -34,19 +39,19 @@ interface MithrilVirtualElement {
 }
 
 interface MithrilModule {
-	controller: Function;
-	view: Function;
+    controller: Function;
+    view: (controller?: any) => MithrilVirtualElement;
 }
 
-interface MithrilDeferred {
-	resolve(value?: any): void;
-	reject(value?: any): void;
-	promise: MithrilPromise;
+interface MithrilDeferred<T> {
+    resolve(value?: T): void;
+    reject(value?: any): void;
+    promise: MithrilPromise<T>;
 }
 
-interface MithrilPromise {
-	(value?: any): any;
-	then(successCallback?: (value: any) => any, errorCallback?: (value: any) => any): MithrilPromise;
+interface MithrilPromise<T> {
+    (value?: T): T;
+    then<R>(successCallback?: (value: T) => R, errorCallback?: (value: any) => any): MithrilPromise<R>;
```

Comment:
I think there's another case to consider here, where the success callback returns another promise:
then<R>(successCallback?: (value: T) => MithrilPromise<R>, errorCallback?: (value: any) => any): MithrilPromise<R>;